react/src/pages/aws/InfrastructurePage.js

```diff
@@ -0,0 +1,1005 @@
+                    <li>No stack outputs.</li>
+                </>}
+            </>:<>
+                <ul style={{marginBottom:"1pt"}}>
+                    { outputs.data && Object.keys(outputs.data)?.map(output => <li key={output}>
+                        <b>{output}</b> <br />
+                        <div style={{wordBreak:"break-all"}}>
+                            { outputs.data[output] === "********" ? <>
+                                <span style={{color:"red"}}>REDACTED</span>
+                            </>:<>
+                                {outputs.data[output]}
+                            </>}
+                        </div>
+                    </li>)}
+                </ul>
+            </>}
+        </div>
+    </div>
+}
+
+const StackParameters = (props) => {
+    const parameters = useFetch(`/aws/stacks/${props.stackName}/parameters`, { cache: true });
+    return <div style={{maxWidth:"480pt"}}>
+        <div className="box lighten">
+            { parameters.empty ? <>
+                { parameters.loading ? <>
+                    <StandardSpinner label="Loading stack parameters" />
+                </>:<>
+                    <li>No stack parameters.</li>
+                </>}
+            </>:<>
+                <ul style={{marginBottom:"1pt"}}>
+                    { parameters.data && Object.keys(parameters.data)?.map(parameter => <li key={parameter}>
+                        <b>{parameter}</b> <br />
+                        <div style={{wordBreak:"break-all"}}>
+                            { parameters.data[parameter] === "********" ? <>
+                                <span style={{color:"red"}}>REDACTED</span>
+                            </>:<>
+                                {parameters.data[parameter]}
+                            </>}
+                        </div>
+                    </li>)}
+                </ul>
+            </>}
+        </div>
+    </div>
+}
+
+const StackResources = (props) => {
+    const resources = useFetch(`/aws/stacks/${props.stackName}/resources`, { cache: true });
+    return <div style={{maxWidth:"480pt"}}>
+        <div className="box lighten">
+            { resources.empty ? <>
+                { resources.loading ? <>
+                    <StandardSpinner label="Loading stack resources" />
+                </>:<>
+                    <li>No stack resources.</li>
+                </>}
+            </>:<>
+                <ul style={{marginBottom:"1pt"}}>
+                    { resources.data && Object.keys(resources.data)?.map(resource => <li key={resource}>
+                        <b style={{wordBreak:"break-all"}}>{resource}</b> <br />
+                        <div style={{wordBreak:"break-all"}}>
+                            { resources.data[resource] === "********" ? <>
+                                <span style={{color:"red"}}>REDACTED</span>
+                            </>:<>
+                                {resources.data[resource]}
```

Comment:
Looks like you have support for this in the client but I'm not sure it will actually happen to any resources in the back-end?

Reply:
Added obfuscation of resources on backend (though not sure really needed, but just in case). The only purpose of the frontend logic is to display a (red) "REDACTED" rather than the stars (********).